### PR TITLE
Make term notation clearer

### PR DESCRIPTION
**Pull request by mpizenberg**
I think these changes for the term documentation we discussed earlier @konstin  make the notation less ambiguous. Let me know what you think. And if that's good we can follow up with your PR to adjust the rule of resolution doc.

### CODE REVIEW

File: src/internals/terms.md [hunk 1]
```diff
@@ -1,11 +1,14 @@
 # Terms
 
+Dependency solving involves handling sets of packages and versions with precision and logic.
+We thus also need precision in our notations.
```
Comment (konstin) [inline src/internals/terms.md:4]: The way i look at it, math gives us powerful tools to solve problems that are in a very specific forms built from few primitives. The problem of dependency management itself is very complex, it has too much complexity to tackle algorithmically directly. So instead, we let the user map them down to cs primitives (e.g. virtual packages for features), map them to math primitives (set and boolean operations), use the solution in math space (CDCL/CDNL), map back to CS primitives and let the user transform them into their output format (e.g. Cargo.lock or requirements.txt)

File: src/internals/terms.md [hunk 2]
```diff
@@ -14,23 +17,41 @@ pub enum Term<V> {
...
 A negative term is evaluated true if and only if
 a version not contained in its range was selected,
 or no version was selected.
-The `negate()` method transforms a positive term into a negative one and vice versa.
+
+> **Remark on negative terms:**
+>
+> A negative term is different than the positive one with its complement range.
+> Indeed, saying "a version was picked in the complement of this range",
+> is different than saying "**no version was picked or** a version was picked in the complement of this range".
```
Comment (konstin) [inline src/internals/terms.md:31]: Thanks!

File: src/internals/terms.md [hunk 3]
```diff
@@ -54,11 +75,13 @@ based on those ranges is defined as follows,
 
 \\[\begin{eqnarray}
 [r_1] \land [r_2] &=& [r_1 \cap r_2],                 \nonumber \\\\
-[r_1] \land \neg [r_2] &=& [r_1 \cap \overline{r_2}], \nonumber \\\\
+[r_1] \land \neg [r_2] &=& [r_1 \cap complement(r_2)], \nonumber \\\\
```
Comment (konstin) [inline src/internals/terms.md:78]: Is that intentionally the text and not a symbol?

**Review by konstin — APPROVED** [review]
Thank you!

File: src/internals/terms.md (hunk 1 above)
Comment (mpizenberg) [reply src/internals/terms.md]: Thanks, I did some changes: https://github.com/pubgrub-rs/guide/pull/11/commits/e0002633992dfd7760f12d7345736d791701f533

File: src/internals/terms.md (hunk 3 above)
Comment (mpizenberg) [reply src/internals/terms.md:82]: Few sentences before, I'm saying that I'd be using overline for terms, and here I use it on a range set, which felt a bit inconsistent. Not sure how to proceed with this one.

Comment (mpizenberg) [reply src/internals/terms.md:82]: Maybe the upper C as showed here would be better https://en.wikipedia.org/wiki/Complement_(set_theory) ?

Comment (mpizenberg) [reply src/internals/terms.md]: image

Is that better? https://github.com/pubgrub-rs/guide/pull/11/commits/37ba8c355db366514a4f0b7ed9533e82022c7bb5

Comment (konstin) [reply src/internals/terms.md]: I like it

**Pull request merged**